tests/Unit/Evolution/Systems/GrMhd/GhValenciaDivClean/BoundaryConditions/Test_ProductOfConditions.cpp

```diff
@@ -0,0 +1,335 @@
+    auto valencia_result = valencia_bc_helper::dg_ghost(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_valencia_condition);
+    auto product_result = product_bc_helper::dg_ghost(
+        make_not_null(&mutable_variables), argument_variables, gridless_box,
+        derived_product_condition);
+    if (gh_result.has_value()) {
+      if (valencia_result.has_value()) {
+        CHECK((product_result.value() ==
+               gh_result.value() + ";" + valencia_result.value()));
+      } else {
+        CHECK(product_result == gh_result);
+      }
+    } else {
+      CHECK(product_result == valencia_result);
+    }
+    CHECK_VARIABLES_APPROX(mutable_variables, expected_mutable_variables);
+  }
+  if constexpr (bc_type ==
+                    evolution::BoundaryConditions::Type::TimeDerivative or
+                bc_type == evolution::BoundaryConditions::Type::
+                               GhostAndTimeDerivative) {
+    auto gh_result = gh_bc_helper::dg_time_derivative(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_gh_condition);
+    auto valencia_result = valencia_bc_helper::dg_time_derivative(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_valencia_condition);
+    auto product_result = product_bc_helper::dg_time_derivative(
+        make_not_null(&mutable_variables), argument_variables, gridless_box,
+        derived_product_condition);
+    if (gh_result.has_value()) {
+      if (valencia_result.has_value()) {
+        CHECK((product_result.value() ==
+               gh_result.value() + ";" + valencia_result.value()));
+      } else {
+        CHECK(product_result == gh_result);
+      }
+    } else {
+      CHECK(product_result == valencia_result);
+    }
+    CHECK_VARIABLES_APPROX(mutable_variables, expected_mutable_variables);
+  }
+  if constexpr (bc_type == evolution::BoundaryConditions::Type::Outflow) {
+    auto gh_result = gh_bc_helper::dg_outflow(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_gh_condition);
+    auto valencia_result = valencia_bc_helper::dg_outflow(
+        make_not_null(&expected_mutable_variables), argument_variables,
+        gridless_box, derived_valencia_condition);
+    auto product_result = product_bc_helper::dg_outflow(
+        make_not_null(&mutable_variables), argument_variables, gridless_box,
+        derived_product_condition);
+    if (gh_result.has_value()) {
+      if (valencia_result.has_value()) {
+        CHECK((product_result.value() ==
+               gh_result.value() + ";" + valencia_result.value()));
+      } else {
+        CHECK(product_result == gh_result);
+      }
+    } else {
+      CHECK(product_result == valencia_result);
+    }
+  }
+}
+}  // namespace
+
+SPECTRE_TEST_CASE(
+    "Unit.Evolution.Systems.GhValenciaDivClean.ProductOfConditions",
```

Comment:
I suggest "Unit.GhValenciaDivClean.BoundaryConditions.ProductOfConditions" for consistency with most systems.